provider/provider.go

```diff
@@ -164,24 +164,26 @@ func (clientTLS *ClientTLS) CreateTLSConfig() (*tls.Config, error) {
 	cert := tls.Certificate{}
 	_, errKeyIsFile := os.Stat(clientTLS.Key)
 
-	if _, errCertIsFile := os.Stat(clientTLS.Cert); errCertIsFile == nil {
-		if errKeyIsFile == nil {
-			cert, err = tls.LoadX509KeyPair(clientTLS.Cert, clientTLS.Key)
-			if err != nil {
-				return nil, fmt.Errorf("Failed to load TLS keypair: %v", err)
+	if len(clientTLS.Cert) > 0 && len(clientTLS.Key) > 0 {
```

Comment:
Thanks for your reply. 

In fact, I suggested to check the `InsecureSkipverify` parameter because, into the test you did, even if the `InsecureSkipVerify` is set to false, the test will be OK but the configuration is not correct (there is no certificate/key file to check) and I would like to avoid this kind of behavior.

The best solution may be to check the `InsecureSkipVerify` and the certificate/key files length before the test you did as described above : 

```go
if ! clientTLS.InsecureSkipVerify && (len(clientTLS.Cert) == 0 || len(clientTLS.Key) == 0) {
        return nil, fmt.Errorf("Certificate and key files must be set when TLS configuration is created") // The message must be improved ;)
}
if len(clientTLS.Cert) > 0 && len(clientTLS.Key) > 0 {
...
}
```

With this new condition, we are sure the TLSConfiguration will be created with no certificate/key file only if the `InsecureSkipVerify` is set to true otherwise, an error will be returned.

@aantono What is your mind about this solution?